chemicaltoolbox/autodock_vina/prepare_box/prepare_box.xml

```diff
@@ -0,0 +1,37 @@
+<tool id="prepare_box" name="Calculate the box parameters for an AutoDock Vina job from an input mol file (confounding box)" version="0.1.0">
+    <requirements>
+        <requirement type="package">rdkit</requirement>
+        <requirement type="package">numpy</requirement>
+    </requirements>
+    <command detect_errors="exit_code"><![CDATA[
+        python $__tool_directory__/calc_vina_box_params.py --ligand '$input1' --config '$output1' --bufx 1 --bufy 2 --bufz 3
+    ]]></command>
+    <inputs>
+        <param type="data" name="input1" format="mol" />
```

Comment:
only mol or also other filetypes are supported?

Reply:
Currently only mol. This could be expanded later, but I think I need a better understanding of how to handle multiple file types with rdkit. I will look for an example in the current tools if we want this before the tool goes into the ToolShed.

Reply:
Galaxy knows the filetype, so we can always pass the filetype into the python script via a parameter if you like. `$input1.ext` for example gives you the Galaxy filetype extension, e.g. (sdf, mol, cml ...)

Reply:
We can write a function very easily to apply the correct rdkit function depending on the file extension, but I guess it should go into a shared library which every rdkit-based tool can access. Not sure of the best way to implement that.

Reply:
Should we create an issue for this? Happy to work with Simon to develop something in the future, if that's something useful we can help to contribute (we use a lot of rdkit stuff, so would be useful for us).

Reply:
Yes, please always helps!

I guess a very tiny package with rdkit-helper functions.

Reply:
I created an issue: https://github.com/bgruening/galaxytools/issues/855

Reply:
Thanks!